numpy/core/src/umath/funcs.inc.src

```diff
@@ -267,6 +267,38 @@ npy_ObjectClip(PyObject *arr, PyObject *min, PyObject *max) {
     return o;
 }
 
+static PyObject *
+npy_ObjectPopCount(PyObject *obj) {
+    PyObject *result;
+
+    /* Try to use inbuilt popcount if available */
+    {
+        static PyObject *builtin_popcount_func = NULL;
+
+        builtin_popcount_func = PyObject_GetAttrString(obj, "bit_count");
+        if (builtin_popcount_func != NULL) {
+            result = PyObject_CallFunction(builtin_popcount_func, NULL);
+            return result;
+        }
+        /* silence errors, and fall back on pure-python `bit_count` */
+        PyErr_Clear();
+    }
+
+    /* Otherwise, use our internal one, written in python */
```

Comment:
I don't like this fallback idea. `bit_count` is defined on Python integers, and as such it is pretty standardized already?  It will work without scalars (since we will implement `bit_count` for those.  I would think the default object fallback should thus be fine?

Reply:
Oh I see. I thought of adding our implementation to support 3.8 and 3.9 as only 3.10 has the internal one and we can eventually remove it. Not sure if this is a standard, we can remove it yeah, what do you think is better?

Reply:
Hmm, that makes sense.  Although I am a bit worried we are adding unnecessary complexity that we can't easily remove later.

Reply:
I was not aware of gcd, it may open up weird corner cases like:
```
np.gcd(np.inf, np.pi, dtype=object)
```
which is an infinite loop.  But I am willing to be convinced that it is worthwhile for some reason.

Reply:
Fixed in 4e78f17, I added a skip in tests as well and now we raise normal error